### DISCUSSION TITLE
Update database topology localUrl and '.fiddler'?

### DISCUSSION THREAD

**Discussion by damccull** [ravendb/ravendb#15474] [Q&A] [answered]
In the GetDatabaseTopologyCommand.java file in the jvm client, there's a bit that checks if the node url contains the string '.fiddler' and, if so, adds it back to the url with the 'localUrl' parameter. Here's the relevant code.
https://github.com/ravendb/ravendb-jvm-client/blob/v5.4/src/main/java/net/ravendb/client/serverwide/commands/GetDatabaseTopologyCommand.java#L45-L49

What is this actually for? I assume it has something to do with the Fiddler http debugging proxy program, but why are there no other references to this that I can find in the code? This is the only location. How is this intended to be used? I can't see anything elsewhere in the library that sets this, so am I missing something or is this perhaps leftover stuff from debugging during the library creation? Is it important to recreate in future libraries?

Thanks.

**Comment by garayx** [comment]
Hi

Yes, this is added to the handle the logic for the `GetDatabaseTopologyCommand` on server side when debugging using fiddler tool.

**Comment by ayende** [comment, marked as answer]
Assume you connect to ravendb using `"http://localhost.fiddler:8080"`
This command is asking the server for the topology, and usually it will reply with `http://localhost:8080` in reply.
Fiddler usually will skip such a request, which means that you cannot easily observe it.

This is meant to ease debugging of communication between client & server under Fiddler.
Note that this is mostly out of favor now, since we are usually using HTTPS, which Fiddler cannot pierce in our scenario.

We now have the traffic watch, which means that this not needed usually.
